pkg/kubernetes/podhelper/service.go

```diff
@@ -0,0 +1,59 @@
+package podhelper
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+	"github.com/openservicemesh/osm-health/pkg/common/outcomes"
+	"github.com/openservicemesh/osm-health/pkg/kuberneteshelper"
+)
+
+// Verify interface compliance
+var _ common.Runnable = (*ServiceCheck)(nil)
+
+// ServiceCheck implements common.Runnable
+type ServiceCheck struct {
+	client kubernetes.Interface
+	pod    *corev1.Pod
+}
+
+// NewServiceCheck checks whether a pod has a corresponding service
+func NewServiceCheck(client kubernetes.Interface, pod *corev1.Pod) ServiceCheck {
+	return ServiceCheck{
+		client: client,
+		pod:    pod,
+	}
+}
+
+// Description implements common.Runnable
+func (check ServiceCheck) Description() string {
+	return fmt.Sprintf("Checking whether destination pod %s has at least one service", check.pod.Name)
+}
+
+// Run implements common.Runnable
+func (check ServiceCheck) Run() outcomes.Outcome {
+	ns := check.pod.Namespace
+	services, err := kuberneteshelper.GetMatchingServices(check.client, check.pod.ObjectMeta.GetLabels(), ns)
+	if err != nil {
+		return outcomes.FailedOutcome{Error: err}
+	}
+	if len(services) == 0 {
+		return outcomes.FailedOutcome{Error: errors.Wrapf(ErrNoService, "destination pod '%s/%s' does not have a corresponding service", ns, check.pod.Name)}
+	}
+
```

Comment:
It'd be nice to log a message here that says - found services A,B,C for pod X.

Reply:
@draychev to clarify do you mean if there are no services found for the pod being checked, it should list generically the services that do exist and their corresponding pods?

Reply:
Tbh I think we should have a way to return a message with the successful outcome. If you thin kit's necessary I can make this change in https://github.com/openservicemesh/osm-health/pull/79 ? 

Alternatively: should this be an Info outcome?

Reply:
I'm thinking we keep it as a fail outcome because it'd be a rare case for the sourcePod to cURL the IP address for the destPod instead of the destPod service